astropy/timeseries/downsample.py

```diff
@@ -31,23 +32,42 @@ def reduceat(array, indices, function):
 
 
 def aggregate_downsample(time_series, *, time_bin_size=None, time_bin_start=None,
-                      n_bins=None, aggregate_func=None):
+                         time_bin_end=None, n_bins=None, aggregate_func=None):
     """
-    Downsample a time series by binning values into bins with a fixed size,
-    using a single function to combine the values in the bin.
+    Downsample a time series by binning values into bins with a fixed size or
+    custom sizes, using a single function to combine the values in the bin.
 
     Parameters
     ----------
     time_series : :class:`~astropy.timeseries.TimeSeries`
         The time series to downsample.
-    time_bin_size : `~astropy.units.Quantity`
-        The time interval for the binned time series.
-    time_bin_start : `~astropy.time.Time`, optional
-        The start time for the binned time series. Defaults to the first
-        time in the sampled time series.
+    time_bin_size : `~astropy.units.Quantity` or `~astropy.time.TimeDelta`, optional
+        The time interval for the binned time series - this is either a scalar
+        value (in which case all time bins will be assumed to have the same
+        duration) or as an array of values (in which case each time bin can
+        have a different duration). If this argument is provided,
+        ``time_bin_end`` should not be provided.
+    time_bin_start : `~astropy.time.Time` or iterable, optional
+        The start time for the binned time series - this can be either given
+        directly as a `~astropy.time.Time` array or as any iterable that
+        initializes the `~astropy.time.Time` class. This can also be a scalar
+        value if ``time_bin_size`` is provided. Defaults to the first
+         time in the sampled time series.
```

Comment:
```suggestion
        time in the sampled time series.
```
I think this was responsible for the
`docstring of astropy.timeseries.downsample.aggregate_downsample:23: WARNING: Unexpected indentation`
in the docs build.

Reply:
Done :) 

Quite a silly mistake - guess I should update my emacs code tests to include more PEP8 conventions.

Reply:
Do you have a flycheck plugin installed? ;-)

Reply:
I do - I followed this more or less https://docs.astropy.org/en/latest/development/codeguide_emacs.html. But somehow it doesn't give me warnings for docstrings (or maybe I don't notice them).